x-pack/plugin/wildcard/src/main/java/org/elasticsearch/xpack/wildcard/mapper/BinaryDvConfirmedAutomatonQuery.java

```diff
@@ -83,8 +84,8 @@ public Scorer scorer(LeafReaderContext context) throws IOException {
                     public boolean matches() throws IOException {
                         if (values.advanceExact(approxDisi.docID()) == false)
                         {
-                            // Bug if we have an indexed value but no doc value.
-                            assert false;
+                            // Bug if we have an indexed value (i.e an approxQuery) but no doc value.
+                            assert approxQuery instanceof MatchAllDocsQuery == false;
```

Comment:
Can you explain why this assertion changed? The original version seemed correct to me.